lib/castle/api/session_sharer.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+require 'singleton'
+
+module Castle
+  module API
+    # this class keeps http session for reuse
```

Comment:
The naming is misleading. We're only caching a `Net::HTTP` object, not an http session.